YAXLib/Deserialization.cs

```diff
@@ -824,10 +824,28 @@ private bool TrySetValueForEmptyElement(object obj, MemberWrapper member, Type m
         // Get the container object from the element (may be null)
         _ = TryGetContainerObject(xElement, collType, memberAlias, out var containerObj);
 
-        var dataItems = new List<object?>(); // this will hold the actual data items
         var collItemType = ReflectionUtils.GetCollectionItemType(collType);
-        var isPrimitive = ReflectionUtils.IsBasicType(collItemType);
 
+        Type? dataItemsType = null;
+        IList dataItems; // this will hold the actual data items
+        if (collItemType == typeof(object))
+        {
+            dataItems = new List<object>();
+        }
+        else
+        {
+            try
+            {
+                dataItemsType = typeof(List<>).MakeGenericType(collItemType);
+                dataItems = (IList) Activator.CreateInstance(dataItemsType)!;
+            }
+            catch (Exception)
```

Comment:
The `Exception` is not covered by a unit test. I'm wondering whether the `try` block would ever throw at this stage of progress? If so, we should cover the case of an `Exception`.

Reply:
I just added it for frameworks where `Activator.CreateInstance` is not supported or similar - but there should be no exception...

Reply:
Ah, I see. All our target frameworks support it, don't they?

Reply:
I'd say: what we can't prove to happen will not happen 😂, so no catch 😉.